delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/FileAttributesUpdateStream.scala

```diff
@@ -0,0 +1,122 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.files
+
+import cats.effect.IO
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.Logger
+import ch.epfl.bluebrain.nexus.delta.kernel.cache.LocalCache
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileRejection._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileState
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.Storages
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.StoragesConfig.RemoteDiskStorageConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.{Storage, StorageRejection, StorageType}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegmentRef
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{ProjectRef, ResourceRef, SuccessElemStream}
+import ch.epfl.bluebrain.nexus.delta.sourcing.offset.Offset
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream._
+import fs2.Stream
+import retry.RetryPolicies.constantDelay
+import retry.implicits.retrySyntaxError
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Stream that attempts to update file attributes asynchronously for linked files in remote storages
+  */
+sealed trait FileAttributesUpdateStream
```

Comment:
Separating from `Files` makes it easier and more consistent with the other long-running tasks (like indexing)